app/routes/jobs/job.py

```diff
@@ -0,0 +1,106 @@
+"""Jobs represent long running analysis tasks. Certain APIs, like querying like
+a list, will return immediately with a job_id. You can poll the job until it's
+complete, and a download like will be provided.
+
+Jobs are only saved for 90 days.
+"""
+import json
+from typing import Any, Dict
+from uuid import UUID
+
+import botocore
+from fastapi import APIRouter, HTTPException, Path
+from fastapi.responses import ORJSONResponse
+
+from ...models.pydantic.user_job import UserJob, UserJobResponse
+from ...settings.globals import RASTER_ANALYSIS_STATE_MACHINE_ARN
+from ...utils.aws import get_sfn_client
+from ..datasets import _get_presigned_url_from_path
+
+router = APIRouter()
+
+
+@router.get(
+    "/{job_id}",
+    response_class=ORJSONResponse,
+    tags=["Jobs"],
+    response_model=UserJobResponse,
+)
+async def get_job(*, job_id: UUID = Path(...)) -> UserJobResponse:
+    """Get job status.
+
+    Jobs expire after 90 days.
+    """
+    try:
+        job = await _get_user_job(job_id)
+        return UserJobResponse(data=job)
+    except botocore.exceptions.ClientError as e:
+        raise HTTPException(status_code=404, detail=str(e))
+
+
+async def _get_user_job(job_id: UUID) -> UserJob:
+    execution = await _get_sfn_execution(job_id)
+
+    if execution["status"] == "SUCCEEDED":
+        output = (
+            json.loads(execution["output"]) if execution["output"] is not None else None
+        )
+
+        if output["status"] == "success":
+            download_link = await _get_presigned_url_from_path(
+                output["data"]["download_link"]
+            )
+            failed_geometries_link = None
+        elif output["status"] == "partial_success":
+            download_link = await _get_presigned_url_from_path(
+                output["data"]["download_link"]
+            )
+            failed_geometries_link = await _get_presigned_url_from_path(
+                output["data"]["failed_geometries_link"]
+            )
+        else:
```

Comment:
Don't you want to also check for 'output["status"] == "failed", in which case the failed_geometries is actually relevant (non-None)?  Both links are None only in the final "error" case you defined.

Reply:
good catch again, added a fix